### PR TITLE
Add a Github Action to build and package fastfetch for Debian.

### PR DESCRIPTION
**Pull request by x-zvf**
This PR adds a github action, which builds fastfetch with all modules enabled (on  Ubuntu), and creates builds a .deb package for easy install on debian systems. The `fastfetch.deb` is automatically added to a release draft.

The action will be useful for catching build errors (such as the `libRpm` typo) as well.

And yes, I'll promise to write better commit messages next time (please quash all of these :D)

### CODE REVIEW

**Comment by x-zvf** [comment]
The Action failed, as it is run in the context of a PR (and can therefore can not create a release draft)

File: .github/workflows/test-and-build-deb-package.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
+    # You can convert this to a matrix build if you need cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    # Install dependencies, so that CMake enables all modules
+    - name: install packages
+      run: sudo apt-get update && sudo apt-get install -y libpci-dev vulkan-tools libvulkan-dev libwayland-client0 libwayland-dev libxrandr-dev libxcb-randr0-dev libdconf-dev libxfconf-0-3 libxfconf-0-dev librpm-dev rpm libzstd-dev libxcb-randr0-dev
```
Comment (LinusDierheimer) [inline .github/workflows/test-and-build-deb-package.yml:24]: `vulkan-tools`, `libwayland-client0`, `libxfconf-0-3`, und `rpm` are not required. `libxcb-randr0-dev` is doubled.

File: .github/workflows/test-and-build-deb-package.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    - name: Upload Release Asset
+      id: upload-release-asset 
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps 
+        asset_path: ${{github.workspace}}/packaging/deb/fastfetch.deb
+        asset_name: fastfetch.deb
+        asset_content_type: application/vnd.debian.binary-package
```
[the remark this answers is not in the document]
Comment (x-zvf) [reply .github/workflows/test-and-build-deb-package.yml:66]: Fair enough, my reasoning was that
a) only a draft release is generated, which can be deleted if the build is not stable
b) No release would be generated if the build failed, obviously (and if it compiles and runs, it is probably not horribly broken). 

I'll split it up into two Actions, with the build of the debian package having to be triggered manually. (I also no longer use debian :D)

[the remark this answers is not in the document]
Comment (LinusDierheimer) [reply .github/workflows/test-and-build-deb-package.yml:66]: The main reason is, that i don't consider fastfetch to be ready to have a release yet. There is still a lot left to do before that.

**Comment by LinusDierheimer** [comment]
Ill merge it and see how this works out for future commits.

**Comment by x-zvf** [comment]
You accidentally merged on a broken commit. A new PR addresses this issue. I am sorry

**Comment by LinusDierheimer** [comment]
Building deb fails with
> [Error : .github#L1](https://github.com/LinusDierheimer/fastfetch/commit/87e410e0fc0c7df340b8cdb9d7684e027a87dabf#annotation_2721689175)
No event triggers defined in `on`

**Comment by x-zvf** [comment]
Add this to the failing script. This is what I said above, you accidentally merged on a broken commit
```
on:
  workflow_dispatch:
```

**Comment by x-zvf** [comment]
Or should I create a new PR?

**Comment by LinusDierheimer** [comment]
Seems to the trick.

**Comment by x-zvf** [comment]
A deb build can be triggered by going to the `actions` tab, clicking on `Select workflow`, then selecting `.github/workflows/build-deb-package.yml` and finally clicking on `run workflow`

**Pull request merged**